Source/SubtitleProvider.h

```diff
@@ -173,17 +171,9 @@ class SubtitleProvider :
         av_packet_free(&packet);
     }
 
-    void SetSubtitleDelay(TimeSpan const& delay)
+    void SetStreamDelay(TimeSpan const& newDelay) override
     {
-        std::lock_guard lock(mutex);
```

Comment:
I think it would be better to set streamDelay immediately here by calling the base method, and do the asynchronous applying of the delay using a separate field ("actualDelay" or whatever). Otherwise, after setting a delay and quickly calling GetStreamDelay(), we might get the old value back.

Reply:
Done.